src/MCPClient/lib/clientScripts/create_transfer_mets.py

```diff
@@ -0,0 +1,351 @@
+
+
+def get_premis_relationship_data(derivations, originals):
+    relationship_data = ()
+
+    for derivation in derivations:
+        relationship_data += (
+            (
+                "relationship",
+                ("relationship_type", "derivation"),
+                ("relationship_sub_type", "is source of"),
+                (
+                    "related_object_identification",
+                    ("related_object_identifier_type", "UUID"),
+                    ("related_object_identifier_value", derivation.derived_file_id),
+                ),
+                (
+                    "related_event_identification",
+                    ("related_event_identifier_type", "UUID"),
+                    ("related_event_identifier_value", derivation.event_id),
+                ),
+            ),
+        )
+    for original in originals:
+        relationship_data += (
+            (
+                "relationship",
+                ("relationship_type", "derivation"),
+                ("relationship_sub_type", "has source"),
+                (
+                    "related_object_identification",
+                    ("related_object_identifier_type", "UUID"),
+                    ("related_object_identifier_value", original.source_file_id),
+                ),
+                (
+                    "related_event_identification",
+                    ("related_event_identifier_type", "UUID"),
+                    ("related_event_identifier_value", original.event_id),
+                ),
+            ),
+        )
+
+    return relationship_data
+
+
+def file_obj_to_premis(file_obj):
+    """
+    Converts an File model object to a PREMIS event object via metsrw.
+
+    Returns:
+        metsrw.plugins.premisrw.premis.PREMISObject
+    """
+
+    premis_digest_algorithm = convert_to_premis_hash_function(file_obj.checksumtype)
+    premis_data = (
+        "object",
+        metsrw.plugins.premisrw.PREMIS_2_2_META,
+        (
+            "object_identifier",
+            ("object_identifier_type", "UUID"),
+            ("object_identifier_value", file_obj.uuid),
+        ),
+        (
+            "object_characteristics",
+            ("composition_level", "0"),
+            (
+                "fixity",
+                ("message_digest_algorithm", premis_digest_algorithm),
+                ("message_digest", file_obj.checksum),
+            ),
+            ("size", str(file_obj.size)),
+            get_premis_format_data(file_obj.fileid_set.all()),
+            (
+                "creating_application",
+                (
+                    "date_created_by_application",
+                    file_obj.modificationtime.strftime("%Y-%m-%d"),
+                ),
+            ),
+        ),
+        ("original_name", escape(file_obj.originallocation)),
+    ) + get_premis_relationship_data(
+        file_obj.related_is_source_of, file_obj.related_has_source
+    )
+
+    return metsrw.plugins.premisrw.data_to_premis(premis_data)
```

Comment:
```suggestion
    return metsrw.plugins.premisrw.data_to_premis(
        premis_data, metsrw.plugins.premisrw.PREMIS_3_0_VERSION)
```

Reply:
Thanks! I was looking at the current behaviour of createprocessedstructmap. Will change.